mdn/developer-portal
# 1223: Design refresh - navigation

This changeset updates the nav to be more like default Mozilla Protocol / mozilla.org

It includes:
* pulling icons from the Page configuration (via the Meta tab) to show beside the link to any child page (ie, all `Topic` pages under "Products & Technologies", plus all children under "Communities"). If there is no icon specified by the page, we use a new default icon.
* including a (fixed) promo link/card in the P&T and Community menus
* Model changes and schema migrations to ensure that all `ContentPage`s and the `People` page can have a topic icon _and_ a description
* A couple of simple Python utils and template tags 
* Adds the new icons to the repo, for reference (hence the large number of files touched in this PR)
* Strips or comments-out CSS which is no longer relevant - it will be purged when we're happy the refresh doesn't need some of it.

It has been cross-browser tested and appears happy in:

- [X] IE 11 Win 10
- [X] Edge Win 10
- [X] FF Win
- [X] FF Mac
- [X] Chrome Win
- [X] Chrome Mac
- [X] Safari Win
- [X] Safari Mac
- [X] Android Chrome (Pixel 3)
- [X] Mobile Safari (iPhone 8)


# To test

It's up on the dev server now and could do wiht a good going-over. Please click links (including the promos to the right-hand side of things) and confirm you're happy

# Screenshots

<img width="448" alt="Screenshot 2020-03-15 at 22 09 20" src="https://user-images.githubusercontent.com/101457/76712163-f45a0f00-670d-11ea-9b81-15f16ab529bb.png">

----

<img width="477" alt="Screenshot 2020-03-15 at 22 09 26" src="https://user-images.githubusercontent.com/101457/76712165-f754ff80-670d-11ea-8e41-e369caa9e78b.png">

----

<img width="468" alt="Screenshot 2020-03-15 at 22 09 31" src="https://user-images.githubusercontent.com/101457/76712166-f7ed9600-670d-11ea-9d7b-4c43dab423f0.png">

----

<img width="411" alt="Screenshot 2020-03-15 at 22 09 45" src="https://user-images.githubusercontent.com/101457/76712168-f8862c80-670d-11ea-9dac-a2a055d233cd.png">

----

<img width="1680" alt="Screenshot 2020-03-15 at 22 36 59" src="https://user-images.githubusercontent.com/101457/76712169-f91ec300-670d-11ea-936f-80e38642b881.png">

----

<img width="1680" alt="Screenshot 2020-03-15 at 22 37 10" src="https://user-images.githubusercontent.com/101457/76712170-f9b75980-670d-11ea-89ec-93552b0affa0.png">

----

<img width="1679" alt="Screenshot 2020-03-15 at 22 37 15" src="https://user-images.githubusercontent.com/101457/76712173-fa4ff000-670d-11ea-86c9-0a8bcb0b431f.png">

----
